cmd/run.go

```diff
@@ -414,7 +414,65 @@ func isSkipScript(hooksGroup, executableName string) bool {
 
 func isSkipCommand(hooksGroup, executableName string) bool {
 	key := strings.Join([]string{hooksGroup, commandsConfigKey, executableName, skipConfigKey}, ".")
-	return viper.GetBool(key)
+
+	value := viper.Get(key)
+
+	switch typedValue := value.(type) {
+	case bool:
+		/*
+			pre-push:
+				commands:
+					packages-audit:
+						skip: true
+		*/
+		return typedValue
+	case string:
+		/*
+			pre-push:
+				commands:
+					packages-audit:
+						skip: merge
+		*/
+		return isSkippedGitState(typedValue)
+	case []interface{}:
+		/*
+			pre-push:
+				commands:
+					packages-audit:
+						skip:
+							- merge
+							- rebase
+		*/
+		for _, gitState := range typedValue {
+			if isSkippedGitState(gitState.(string)) {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+func isSkippedGitState(state string) bool {
+	return state == "merge" && isMergeInProgress() || state == "rebase" && isRebaseInProgress()
+}
+
+func isMergeInProgress() bool {
+	if _, err := os.Stat(filepath.Join(getRootPath(), ".git", "MERGE_HEAD")); os.IsNotExist(err) {
+		return false
+	}
+
+	return true
+}
+
+func isRebaseInProgress() bool {
+	if _, mergeErr := os.Stat(filepath.Join(getRootPath(), ".git", "rebase-merge")); os.IsNotExist(mergeErr) {
+		if _, applyErr := os.Stat(filepath.Join(getRootPath(), ".git", "rebase-apply")); os.IsNotExist(applyErr) {
```

Comment:
I wonder, whether can we check for single `REBASE_HEAD` file instead of this two checks? :thinking:

Also here we need to use output of `git rev-parse --git-dir` command to get `.git` directory for worktrees.

Reply:
[Looks like](https://git-scm.com/docs/git-rebase#Documentation/git-rebase.txt---show-current-patch) REBASE_HEAD exists only when it's an "interactive rebase or when rebase is stopped because of conflicts"

Reply:
Yeah, official git prompts also checks for these directories, so let's keep it as it is: https://github.com/git/git/blob/de88ac70f3a801262eb3aa087e5d9a712be0a54a/contrib/completion/git-prompt.sh#L446-L462